sample_apps/simple_app/run_example.sh

```diff
@@ -0,0 +1,638 @@
+#!/bin/bash
+# ##############################################################################
+# run_example.sh - Driver script to build things needed to run sample-apps
+# ##############################################################################
+
+set -Eeuo pipefail
+
+# Setup script globals, to establish curdir and root path to Certifier code base
+Me=$(basename "$0")
+
+
+# Establish full dirpath where this script lives
+pushd "$(dirname "$0")" > /dev/null 2>&1
+EXAMPLE_DIR=$(pwd); export EXAMPLE_DIR
+popd > /dev/null 2>&1
+
+Cleanup="${EXAMPLE_DIR}/cleanup.sh"
+PROV_DIR="${EXAMPLE_DIR}/provisioning"
+
+# shellcheck disable=SC2046
+CERT_PROTO="$(dirname $(dirname "$EXAMPLE_DIR"))"; export CERT_PROTO
+CERT_UTILS="${CERT_PROTO}/utilities"
+ExName=$(basename "$EXAMPLE_DIR")
+
+MeMsg="Build and run example program ${ExName}"
+
+# Script global symbols to app-data dirs
+Client_app_data="./app1_data/"
+Srvr_app_data="./app2_data/"
+
+SimpleServer_PID=0
+AppAsServer_PID=0
+Found_step=0        # While looking thru Steps[] for user-specified fn-name
+
+# ###########################################################################
+# Set trap handlers for all errors. Needs -E (-o errtrace): Ensures that ERR
+# traps (see below) get inherited by functions, command substitutions, and
+# subshell environments
+# Ref: https://citizen428.net/blog/bash-error-handling-with-trap/
+#
+function cleanup() {
+    set +x
+    echo "${Me}: Failed command, ${BASH_COMMAND}, at line $(caller) while executing function ${This_fn}"
+
+    do_cleanup
+}
+
+# ###########################################################################
+# Calls the cleanup sub-script, which cleans up and returns 0 (always)
+# This avoids this script failing with non-zero $rc when stale stuff is cleaned up.
+function do_cleanup() {
+    ${Cleanup} $$
+}
+trap cleanup ERR
```

Comment:
All this and function `cleanup()` and the `set` on L6 at the top is to ensure that any error is trapped and the script will abort immediately.

Messy stuff ... till I got it working ... somewhat stably.